gravitee-am-cypress/cypress/integration/oidc/ciba/ciba.spec.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+    createAcceptAllDeviceNotifier,
+    createDomain,
+    requestDomain,
+    startDomain
+} from "../../../commands/management/domain-management-commands";
+import {domainOidcCibaConfig, oidcApplication} from "../../../fixtures/oidc";
+import {User} from "../../../model/users";
+import {requestAdminAccessToken} from "../../../commands/management/token-management-commands";
+import {createUser} from "../../../commands/management/user-management-commands";
+
+const configureDomainForCiba = (domain, accessToken, method, body) =>
+    requestDomain(domain, accessToken, "PATCH", body)
+        .should(response => expect(response.status).to.eq(200));
+
+const createCibaApplication = (domainHrid) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/register`,
+    method: 'POST',
+    retryOnStatusCodeFailure: true,
+    body: oidcApplication
+}).should(response => expect(response.status).to.eq(201));
+
+const registerDomainWithNotifier = (domain, domainHrid, clientId, clientSecret) => cy.request({
+    url: Cypress.env().cibaDelegateNotifierUrl + "/domains",
+    method: 'POST',
+    body: {
+        domainId: domain,
+        domainCallback: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate/callback`,
+        clientId: clientId,
+        clientSecret: clientSecret
+    }
+}).should(response => expect(response.status).to.eq(200))
+
+const requestAuthentication = (domainHrid, clientId, clientSecret, loginHint, userCode) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate`,
+    method: 'POST',
+    auth: {
+        user: clientId,
+        password: clientSecret
+    },
+    form: true,
+    body: {
+        scope: "openid",
+        login_hint: loginHint,
+        user_code: userCode
+    },
+    failOnStatusCode: false
+})
+
+describe("An application is correctly configured to use CIBA", () => {
+    let accessToken;
+    let domain;
+    let client;
+    let auth_req_id;
+    let cibaConfig;
+    const user: User = {
+        firstName: "Pat",
+        lastName: "Test",
+        email: "pat@test.com",
+        username: "pat@test.com",
+        password: "Password123",
```

Comment:
Would it be easier to reuse this username and password in other tests if we declare it in fixtures?